### DISCUSSION TITLE
useFetch's refresh method successfully fetching new data, but not rendering

### DISCUSSION THREAD

**Discussion by jader-rubini** [nuxt/nuxt#23151] [Questions]
Hey folks! Newbie question here…

I've managed to successfully fetch data from an external API with the following code:
```
<script setup>
const { data, pending, refresh } = await useFetch(() => 'https://api.adviceslip.com/advice')
const { slip } = JSON.parse(data.value)
</script>

<template>
[...]
  <advice-slip :slip="slip" :key="slip.id" />
  <refresh-button @click="refresh" />
[...]
</template>
```
Turns out when I click the refresh button, If I open the Network tab on my devtools, I can see the new request being made and the new data being fetched, but the template won't update with the new data. What am I doing wrong?

**Also, bonus question:** wasn't `useFetch` supposed to automatically parse the response data into an object? I wrestled with this request an embarrassing amount of time before realizing I was getting a plain string as a response and had to parse it myself.

**Comment by logotip4ik** [comment]
Hello! Well, your api return `Content-Type` header with value `text/html; charset=UTF-8` which means that response is html, not json, that's why `useFetch` did not parse it. Also, `const { slip } = JSON.parse(data.value)` at this line you lose reactivity. You need to wrap it into `computed` to catch dependencies and update return value based on deps. But i think better solution would be to use `transform` property in `useFetch` and pass it `JSON.parse` as value.

[Working demo](https://stackblitz.com/edit/nuxt-starter-9axuca?file=app.vue)
[`useFetch` docs](https://nuxt.com/docs/api/composables/use-fetch#usefetch)
